src/agentlab2/agents/react.py

```diff
@@ -0,0 +1,148 @@
+import logging
+
+from termcolor import colored
+
+from agentlab2.agent import Agent, AgentConfig
+from agentlab2.core import ActionSchema, AgentOutput, Observation
+from agentlab2.llm import LLM, Prompt, obs_to_messages
+
+logger = logging.getLogger(__name__)
+
+
+class ReactAgentConfig(AgentConfig):
+    llm: LLM
+    use_html: bool = True
+    use_axtree: bool = False
+    use_screenshot: bool = True
+    max_actions: int = 10
+    max_obs_chars: int = 100000  # truncate long observations to M chars
+    max_history_tokens: int = 120000  # compact history if it exceeds N tokens
+    system_prompt: str = """
+You are an expert AI Agent trained to assist users with complex web tasks.
+Your role is to understand the goal, perform actions until the goal is accomplished and respond in a helpful and accurate manner.
+Keep your replies brief, concise, direct and on topic. Prioritize clarity and avoid over-elaboration.
+Do not express emotions or opinions."""
+    react_prompt: str = """
+Think along the following lines:
+1. Summarize the last observation and describe the visible changes in the state.
+2. Evaluate action success, explain impact on task and next steps.
+3. If you see any errors in the last observation, think about it. If there is no error, just move on.
+4. List next steps to move towards the goal and propose next immediate action.
+Then produce the single function call that performs the proposed action. If the task is complete, produce the final step."""
+    summarize_system_prompt: str = """
+You are a helpful assistant that summarizes agent interaction history. Following messages is the history to summarize:"""
+    summarize_prompt: str = """
+Summarize the presented agent interaction history concisely.
+Focus on:
+- The original goal
+- Key actions taken and their outcomes
+- Important errors or obstacles encountered
+- Current progress toward the goal
+Provide a concise summary that preserves all information needed to continue the task."""
+
+    def make(self, actions: list[ActionSchema]) -> "ReactAgent":
+        return ReactAgent(config=self, actions=actions)
+
+
+class ReactAgent(Agent):
+    metadata = {
```

Comment:
I think all these propriety should be officially part of the base class Agent and not a metadata dict, but official properties